src/Akka.Persistence.SqlServer/Helpers/ColumnSizeLoader.cs

```diff
@@ -0,0 +1,87 @@
+// //-----------------------------------------------------------------------
+// // <copyright file="ColumnSizeLoader.cs" company="Akka.NET Project">
+// //     Copyright (C) 2009-2021 Lightbend Inc. <http://www.lightbend.com>
+// //     Copyright (C) 2013-2021 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// // </copyright>
+// //-----------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Common;
+using System.Linq;
+using Akka.Persistence.Sql.Common.Journal;
+
+namespace Akka.Persistence.SqlServer.Helpers
+{
+    /// <summary>
+    /// Helper class that can load column sizes from SqlServer table
+    /// </summary>
+    internal static class ColumnSizeLoader
+    {
+        /// <summary>
+        /// Loads column sizes for Journal
+        /// </summary>
+        /// <returns></returns>
+        public static JournalColumnSizesInfo LoadJournalColumnSizes(QueryConfiguration conventions, DbConnection connection)
+        {
+            using (var command = connection.CreateCommand())
+            {
+                connection.Open();
+                command.CommandText = $"SELECT * FROM {conventions.FullJournalTableName}";
+
+                // start reading - no need to load the table's content
+                using (var reader = command.ExecuteReader())
+                {
+                    // load columns metadata
+                    var results = LoadSchemaTableInfo(reader);
+
+                    return new JournalColumnSizesInfo(
+                        persistenceIdColumnSize: (int)results.First(r => r["ColumnName"].ToString() == conventions.PersistenceIdColumnName)["ColumnSize"],
+                        tagsColumnSize: (int)results.First(r => r["ColumnName"].ToString() == conventions.TagsColumnName)["ColumnSize"],
+                        manifestColumnSize: (int)results.First(r => r["ColumnName"].ToString() == conventions.ManifestColumnName)["ColumnSize"]
```

Comment:
This is a SQLServer-specific code, which uses the database schema table to extract columns metadata. In this case, `ColumnSize` is what we need.